File: Code/Core/Revenj.Serialization/PassThroughSerialization.cs
using System.Runtime.Serialization;

namespace Revenj.Serialization
{
	public class PassThroughSerialization : ISerialization<object>
	{
		public object Serialize<T>(T value) { return value; }

		public T Deserialize<T>(object data, StreamingContext context)
		{
			return (T)data;
		}
	}
}

File: java/src/main/java/org/revenj/serialization/PassThroughSerialization.java
package org.revenj.serialization;

import org.revenj.patterns.Serialization;

import java.io.IOException;
import java.lang.reflect.Type;

final class PassThroughSerialization implements Serialization<Object> {

	@Override
	public Object serialize(Object value) {
		return value;
	}

	@Override
	public Object deserialize(Type type, Object data) throws IOException {
		return data;
	}
}
